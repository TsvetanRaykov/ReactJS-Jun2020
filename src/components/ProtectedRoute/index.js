import React from 'react'
import { Route, Redirect } from 'react-router-dom'
import userService from '../../services/userService'

const ProtectedRoute = ({ component: Component, ...rest }) => {
	const user = userService.getCurrentUser()
	return (
		<Route
			{...rest}
			render={(props) => {
				if (user) {
					return <Component {...rest} {...props} />
				} else {
					return <Redirect to='/login' />
				}
			}}
		/>
	)
}

export default ProtectedRoute
