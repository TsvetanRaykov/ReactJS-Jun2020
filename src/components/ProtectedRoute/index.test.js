import React from 'react'
import { act } from 'react-test-renderer'
import { shallow } from 'enzyme'
import TestWrapper from '../../utils/test-wrapper'
import ProtectedRoute from '.'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<ProtectedRoute />', () => {
	it('Should match the snapshot', () => {
		act(() => {
			const wrapper = shallow(
				<TestWrapper>
					<ProtectedRoute exact path='/' component={<h1>TEST</h1>}>
						<h1>TEST</h1>
					</ProtectedRoute>
				</TestWrapper>
			)
			expect(wrapper).toMatchSnapshot()
		})
	})
})
