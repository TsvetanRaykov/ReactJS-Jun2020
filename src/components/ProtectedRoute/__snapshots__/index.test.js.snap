// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProtectedRoute /> Should match the snapshot 1`] = `ShallowWrapper {}`;
